include/sos/barriers/LHSCB.h

```diff
@@ -0,0 +1,89 @@
+// VolEsti (volume computation and sampling library)
+//
+// Copyright (c) 2020 Bento Natura
+//
+// Licensed under GNU LGPL.3, see LICENCE file
+
+#ifndef NONSYMMETRICCONICOPTIMIZATION_LHSCB_H
+#define NONSYMMETRICCONICOPTIMIZATION_LHSCB_H
+
+#include<iostream>
+#include <vector>
+#include "../utils.h"
+
+template <typename IPMDouble>
+class LHSCB {
+    typedef Vector<IPMDouble> Vector;
+    typedef Matrix<IPMDouble> Matrix;
+public:
+    LHSCB() : _num_variables(0) {};
+
+    template <typename T>
+    void cast_members_from(const LHSCB<T> & other){
+       _num_variables = other._num_variables;
+    }
+
+    template<typename T>
+    LHSCB<T> * cast(){
+        LHSCB<T> * lhscb;
+        (*lhscb).template cast_members_from<IPMDouble>(*this);
+        return lhscb;
+    }
+
+    virtual ~LHSCB() {};
```

Comment:
please eliminate the use of virtual functions

Reply:
It is very natural to use virtual function. The Skajaa-Ye algorithm needs a Logarithmically-Homogeneous-Self-Concordant-Barrier (LHSCB) as input, which has to implement several functions. This could be the Interpolant Barrier for the dual SOS cone that we mostly use or any other barrier for the SOS cone or other conoes like the non-negative orthant or the PSD cone.
Can you please elaborate why we should implement this differently, considering that it is not a bottleneck in computation?

Reply:
It's a matter of design, it is not good practice to mix designs in a library. If it is difficult to change and since this part is not interacting a lot with the rest of the library I am OK to leave as is. In any case my main concern is performance.